pilot/pkg/serviceregistry/kube/controller/endpointcontroller.go

```diff
@@ -109,6 +110,15 @@ func updateEDS(c *Controller, epc kubeEndpointsController, ep interface{}, event
 		}
 	}
 
+	// Handle external address update for NodePort services whose external
```

Comment:
Would we enter this func for these updates (do NodePort services update Endpoints/EndpointSlice)?

Reply:
Why do we need to handle node port service when endpoint update?

Reply:
@stevenctl @hzxuzhonghu - No. `NodePort` services don't update Endpoints/EndpointSlice. But when external traffic policy is Local, we need to handle endpoint update. Let me give you an example

Imagine a cluster with 3 nodes on which we can schedule gateway pods (let us call them node-1, node-2 and node-3). Let node's external IPs be 
```
node-1: 34.34.78.78
node-2: 34.34.78.79
node-3: 34.34.78.80
```
Now let us assume that the pod is initially scheduled in node-2. So the list of external addresses would only have `34.34.78.79`. When the pod is rescheduled to node-1 (say, because node-2 is being upgraded/patched), the list of external addresses should be `34.34.78.78`. 

As you can see, the list of external addresses depends on the nodes where endpoints are scheduled. So we should also handle endpoint events to update the mapping from service-hostname to the list of nodes containing the workload. In the example, I gave it would be initially `istio-eastwestgateway -> [node-2]`. On endpoint event due to reschedule, it would be `istio-eastwestgateway -> [node-1]`. The maps are updated by `updateServiceNodePortAddresses` called due to endpoint event.

Reply:
When our set of node addresses change, this will trigger us to recompute the mappings:

https://github.com/istio/istio/blob/943ba0765876590d5c6da89d5df034fe0ea0808a/pilot/pkg/serviceregistry/kube/controller/controller.go#L531

https://github.com/istio/istio/blob/943ba0765876590d5c6da89d5df034fe0ea0808a/pilot/pkg/serviceregistry/kube/controller/controller.go#L456

Assuming the correct set of addresses are updated here:

https://github.com/istio/istio/blob/1fe1ec39d19502ebef06b1c1e22daba213175f30/pilot/pkg/serviceregistry/kube/controller/network.go#L159

I think everything is compatible with that already.